### PR TITLE
Cull pre-pluralised strings when pruning

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/vector-im/element-web/issues/22349

### CODE REVIEW

File: scripts/gen-i18n.ts [hunk 1]
```diff
@@ -142,36 +149,35 @@ function getTranslationsJs(file) {
...
 
                     // check the format string against the args
                     // We only check _t: _td has no args
-                    if (node.callee.name === '_t') {
+                    if (isIdentifier(p.node.callee) && p.node.callee.name === '_t') {
                         try {
                             const placeholders = getFormatStrings(tKey);
-                            for (const placeholder of placeholders) {
-                                if (node.arguments.length < 2) {
+                            for (const placeholder of Array.from(placeholders)) {
```
Comment (robintown) [inline scripts/gen-i18n.ts]: I don't think the `Array.from` should be necessary, a for can take any iterable

File: scripts/gen-i18n.ts [hunk 2]
```diff
@@ -276,7 +285,7 @@ const walkOpts = {
                 return;
             }
             console.log(`${fullPath} (${trs.size} strings)`);
-            for (const tr of trs.values()) {
+            for (const tr of Array.from(trs)) {
```
Comment (robintown) [inline scripts/gen-i18n.ts]: Same here

File: scripts/gen-i18n.ts [hunk 3]
```diff
@@ -286,12 +295,12 @@ const walkOpts = {
...
     if (fs.existsSync(path)) {
-        walk.walkSync(path, walkOpts);
+        walkSync(path, walkOpts);
     }
 }
 
-const trObj = {};
-for (const tr of translatables) {
+const trObj: Record<string, string | string[]> = {};
+for (const tr of Array.from(translatables)) {
```
Comment (robintown) [inline scripts/gen-i18n.ts]: Same here

**Review by robintown — APPROVED** [review]
Looks fine otherwise, thanks for doing the TS conversion

File: scripts/gen-i18n.ts (hunk 1 above)
Comment (t3chguy) [reply scripts/gen-i18n.ts]: Good spot, my local TS was complaining about needing to be set to es2015 or higher for that to work, even though its set to es2016 but will try again

Comment (t3chguy) [reply scripts/gen-i18n.ts]: Just my IDE is unhappy, tsc seems happy

**Comment by andybalaam** [comment]
@t3chguy you are my hero

**Pull request merged**